src/lib/prep.ml

```diff
@@ -191,48 +188,33 @@ module Prep = struct
       match Storage.parse_hash ~prefix:"HASHES" line with
       | Some value -> (value :: git_hashes, failed, retriable_errors)
       | None -> (
+        if retry_conditions line then
+          (git_hashes, failed, line :: retriable_errors)
```

Comment:
@TheLortex this should address the issue you pointed out about the check only being applied to `prev`